src/storr.c

```diff
@@ -0,0 +1,34 @@
+#include <R.h>
+#include <R_ext/Rdynload.h>
+#include <Rinternals.h>
+#include <Rversion.h>
+
+SEXP Cread_text_file(SEXP path, SEXP nchar) {
+  FILE *fp;
+  fp = fopen(CHAR(asChar(path)), "rb");
+  if (fp == NULL) {
+    return R_NilValue;
+  }
+  int n = asInteger(nchar) + 1; // Need an extra character for '\0'.
+  char *buf = (char*) malloc(n * sizeof(char));
+  fgets(buf, n, fp);
+  fclose(fp);
+  SEXP out = PROTECT(allocVector(STRSXP, 1));
```

Comment:
I think that `out = PROTECT(mkString(buf));` here will work and be replace these two lines

Reply:
474247f001104c53fa0542598ea25a4c5a20b4ae